src/Entity/Auth/UserProvider.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace App\Entity\Auth;
+
+use Cocur\Slugify\SlugifyInterface;
+use Doctrine\ORM\EntityManagerInterface;
+use HWI\Bundle\OAuthBundle\OAuth\Response\UserResponseInterface;
+use HWI\Bundle\OAuthBundle\Security\Core\User\OAuthAwareUserProviderInterface;
+use Symfony\Component\Security\Core\Exception\UnsupportedUserException;
+use Symfony\Component\Security\Core\Exception\UsernameNotFoundException;
+use Symfony\Component\Security\Core\User\UserInterface;
+use Symfony\Component\Security\Core\User\UserProviderInterface;
+
+class UserProvider implements UserProviderInterface, OAuthAwareUserProviderInterface
+{
+    /**
+     * @var EntityManagerInterface
+     */
+    protected $entityManager;
+
+    /**
+     * @var SlugifyInterface
+     */
+    protected $slugify;
+
+    /**
+     * UserProvider constructor.
+     *
+     * @param EntityManagerInterface $entityManager
+     * @param SlugifyInterface       $slugify
+     */
+    public function __construct(EntityManagerInterface $entityManager, SlugifyInterface $slugify)
```

Comment:
Hmm what about injecting `UserRepository` directly?

Reply:
Since I'm using both `UserRepository` & `UserCharacterRepository` and also using the `EntityManagerInterface` itself to persists some object I'm creating.
I think it's lighter this way, but since it's the first UserProvider I made it might not be the right way to do this ?

FYI: The OAuth will return me some `UserCharacter` related-data (and not related to `User`), so I need to link this `UserCharacter` to the correct `User` or create it if it doesn't exists.

Reply:
@Korbeil the idea is to segregate this even further If you inject all repositories that you need for this to work you can then understand what areas this service requires. I think it will help you on a long term, and is so easy when you make them auto-wireable

Reply:
After talking on Symfony slack with @tarlepp we agree that even persist part can be managed by the `Repository`. So I think I'll create an `AbstractRepository` to make this more generic (for all my repositories).

I'm more worried about the way to do it. Is it "good design" to make theses as `persist(object $entityObject)` function directly or as some custom function like `create(array $entityDetails)` that will take only an array with all values and will make the persist internally ?

Reply:
After reading: https://matthiasnoback.nl/2014/05/inject-a-repository-instead-of-an-entity-manager/ (linked by @tarlepp). I'll do with a `save(object)` method as it's done in the article ;)

Thanks for helping here !